### PR TITLE
add i18n for core components

### PR DESCRIPTION
**Pull request by mario-mui**
#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by vinzscam — APPROVED** [review]
amazing 👏

**Comment by Rugvip** [comment]
Linking this one to #21267 as well

**Comment by mario-mui** [comment]
@vinzscam @tikabom @Rugvip please review this PR, I had fix all unit test

**Review by tikabom — APPROVED** [review]

**Comment by mario-mui** [comment]
@tikabom PR not merge automatically, is this need any other reviewer to approve ?

**Comment by emmaindal** [comment]
hello @mario-mui, it would require approval from @backstage/maintainers before its possible to merge this in, in the meanwhile can you rebase this on the latest master to resolve the conflicts. Thank you!

**Comment by benjdlambert** [comment]
Hey @mario-mui!

Thanks for the PR! As you might have seen there's still some ongoing discussion and decisions happening over in https://github.com/backstage/backstage/issues/21267 that we'd like to settle before merging the i18n PRs. We're hoping to get these documented this week so that we can unblock these i18n PR's and get them merged :tada:

**Comment by mario-mui** [comment]
@benjdlambert @Rugvip I had change messages keys to nested and camelCase

**Review by agentbellnorm — APPROVED** [review]
techdocs changes look good 👍

**Comment by mario-mui** [comment]
@benjdlambert This is ready for merge?

**Comment by benjdlambert** [comment]
@mario-mui not yet unfortunately, I think that we might just need to possibly make some small changes to the keys and their nesting shape. I'm hoping to have some docs and recommendations for i18n stuff written up by the end of the day!

**Comment by mario-mui** [comment]
message: {
  componentName: {
      key1:
      key2
  }
}
what about this translation structure？ @benjdlambert

**Comment by benjdlambert** [comment]
Hey @mario-mui :wave:

I don't know if you saw but there's some updated documentation now https://backstage.io/docs/plugins/internationalization#guidelines-for-i18n-messages-and-keys with some suggestions for how to structure the key names.

I think `componentName` could be a little too specific, think more about the context and where it is, but I think it depends.

What would be an example?

**Comment by mario-mui** [comment]
for instance: https://github.com/backstage/backstage/pull/20932/files#diff-fd24c3513000f2c84e0f9310bcb6efa282b69c472b38fd73b1088d15330bf66dR117

The `customProvider` is underlie `src.layout.SignInPage` so `tokenInvalid` key should defined like ?
```
layout: {
  signInPage: {
     tokenInvalid: 'Token is not a valid OpenID Connect JWT Token'
  }
}
```
@benjdlambert

**Comment by mario-mui** [comment]
> for instance: https://github.com/backstage/backstage/pull/20932/files#diff-fd24c3513000f2c84e0f9310bcb6efa282b69c472b38fd73b1088d15330bf66dR117
> 
> The `customProvider` is underlie `src.layout.SignInPage` so `tokenInvalid` key should defined like ?
> 
> ```
> layout: {
>   signInPage: {
>      tokenInvalid: 'Token is not a valid OpenID Connect JWT Token'
>   }
> }
> ```
> 
> @benjdlambert

Hi, @benjdlambert are you have time to have a look

**Comment by benjdlambert** [comment]
@mario-mui sorry just getting back up to speed after the holidays.

I wouldn't pay too much attention to the filepath and filenames as those can change frequently and then updating the `i18n` messages would be unfortunate side effects.

I think for your example, focus on more of the context, what the component does and what is the message purpose. You can drop the `layout` I would say.

Something like `signIn.invalidToken` or something like that makes a little more sense to me.

**Comment by mario-mui** [comment]
> @mario-mui sorry just getting back up to speed after the holidays.
> 
> I wouldn't pay too much attention to the filepath and filenames as those can change frequently and then updating the `i18n` messages would be unfortunate side effects.
> 
> I think for your example, focus on more of the context, what the component does and what is the message purpose. You can drop the `layout` I would say.
> 
> Something like `signIn.invalidToken` or something like that makes a little more sense to me.

@benjdlambert can you review this PR again

**Comment by mario-mui** [comment]
@benjdlambert @Rugvip Sorry to disturb you, since there is no new comment or other thing happen to this PR for two weeks.

I want to know, is this PR being review or something else need me to do ?

**Comment by mario-mui** [comment]
> @mario-mui sorry we've been pretty slammed with a lot of work that last few weeks, so sorry it's taking a while. This is on our TODO to sit down a review it. We will get to it soon! 🎉

I see, thanks

File: packages/core-components/api-report-alpha.md
Comment (benjdlambert) [inline packages/core-components/api-report-alpha.md:19]: Think that we should condense these keys if possible with a `\n` or `<br/>` in the middle. Think that these should be one message rather than splitting it up into two keys.

File: packages/core-components/api-report-alpha.md
Comment (benjdlambert) [inline packages/core-components/api-report-alpha.md:28]: `skipToContent`

File: packages/core-components/api-report-alpha.md
Comment (benjdlambert) [inline packages/core-components/api-report-alpha.md:29]: Think we can skip this, and `sidebar.recentlyViewedIntroText` as we're going to clean them up, they're really not used, and should belong elsewhere rather than `core-components`

File: packages/core-components/api-report-alpha.md
Comment (benjdlambert) [inline packages/core-components/api-report-alpha.md:43]: Can we revert these and just keep the text inline for now. We want to support interpolating React components which is going to help keep these messages a little cleaner and just having 2 nested ones.

File: packages/core-components/api-report-alpha.md
Comment (benjdlambert) [inline packages/core-components/api-report-alpha.md:50]: Same as above, think we want to fix the interpolation to make these messages cleaner. Just leave the translations in the component as before.

File: packages/core-components/src/components/AlertDisplay/AlertDisplay.tsx
Comment (benjdlambert) [inline packages/core-components/src/components/AlertDisplay/AlertDisplay.tsx:128]: Im pretty sure that in the translation string you should be able to do `{{ count }}` instead of having two variables for this. Please let us know if this doesn't work :)

File: packages/core-components/api-report-alpha.md
Comment (benjdlambert) [inline packages/core-components/api-report-alpha.md:66]: Same as above again, this would be better solved with interpolation rather than seperate messages.

File: packages/core-components/src/hooks/useSupportConfig.ts
Comment (benjdlambert) [inline packages/core-components/src/hooks/useSupportConfig.ts:43]: `supportConfig.default.title`?

File: packages/core-components/src/hooks/useSupportConfig.ts
Comment (benjdlambert) [inline packages/core-components/src/hooks/useSupportConfig.ts:48]: `supportConfig.default.linkTitle`?

File: packages/core-components/src/layout/SignInPage/guestProvider.tsx
Comment (benjdlambert) [inline packages/core-components/src/layout/SignInPage/guestProvider.tsx:50]: Think we can condense these two into one key with a `\n` or `<br/>` in the middle whichever one works.

**Review by benjdlambert — CHANGES_REQUESTED** [review]
This is looking great, just a few little things that we want to iron out here too.

**Comment by mario-mui** [comment]
> This is looking great, just a few little things that we want to iron out here too.

Cool, I will fix this today

File: packages/core-components/api-report-alpha.md
Comment (mario-mui) [reply packages/core-components/api-report-alpha.md:28]: done

File: packages/core-components/api-report-alpha.md
Comment (mario-mui) [reply packages/core-components/api-report-alpha.md:29]: ok

File: packages/core-components/api-report-alpha.md
Comment (mario-mui) [reply packages/core-components/api-report-alpha.md:43]: ok, I had revert `orPlease ` `contactSupport ` `isBug `

File: packages/core-components/src/hooks/useSupportConfig.ts
Comment (mario-mui) [reply packages/core-components/src/hooks/useSupportConfig.ts:43]: done

File: packages/core-components/src/hooks/useSupportConfig.ts
Comment (mario-mui) [reply packages/core-components/src/hooks/useSupportConfig.ts:48]: done

File: packages/core-components/src/components/AlertDisplay/AlertDisplay.tsx
Comment (mario-mui) [reply packages/core-components/src/components/AlertDisplay/AlertDisplay.tsx:128]: done

File: packages/core-components/api-report-alpha.md
Comment (mario-mui) [reply packages/core-components/api-report-alpha.md:19]: done

File: packages/core-components/src/layout/SignInPage/guestProvider.tsx
Comment (mario-mui) [reply packages/core-components/src/layout/SignInPage/guestProvider.tsx:50]: done

**Comment by benjdlambert** [comment]
@mario-mui I think you might want to rebase this to fix the conflict and the e2e errors :)

**Comment by mario-mui** [comment]
> @mario-mui I think you might want to rebase this to fix the conflict and the e2e errors :)

sure, I just come from Chinese new year holiday. I will fix it soon

**Comment by mario-mui** [comment]
> @mario-mui I think you might want to rebase this to fix the conflict and the e2e errors :)

Done. @benjdlambert Ready to merge?

File: packages/core-components/api-report-alpha.md
Comment (benjdlambert) [reply packages/core-components/api-report-alpha.md:60]: You can remove these translations, and revert the components until we fix the interpolation feature. Same for `emptyState.missingAnnotation.descriptionPrefix*` ones

**Comment by benjdlambert** [comment]
@mario-mui I think that we still need to address these: https://github.com/backstage/backstage/pull/20932#discussion_r1494484164

**Comment by mario-mui** [comment]
> @mario-mui I think that we still need to address these: [#20932 (comment)](https://github.com/backstage/backstage/pull/20932#discussion_r1494484164)

**Comment by mario-mui** [comment]
> @mario-mui I think that we still need to address these: [#20932 (comment)](https://github.com/backstage/backstage/pull/20932#discussion_r1494484164)

Had revert some suffix translation keys.

**Comment by mario-mui** [comment]
@benjdlambert Hi, is this PR now ready to merge ?

File: packages/core-components/api-report-alpha.md
Comment (benjdlambert) [inline packages/core-components/api-report-alpha.md:26]: `skipToContent`

**Review by Rugvip — APPROVED** [review]
Think this looks good now! 🎉 👏 

Thank you @mario-mui, let's :shipit:!

**Pull request merged**
